### ISSUE TITLE
job[ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge] does not exist

### ISSUE THREAD

**Issue by djgalloway** [ceph/mita#129]
```
[2019-08-13 19:46:04,394: INFO/ForkPoolWorker-3] found stuck task with name: ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge
[2019-08-13 19:46:04,394: INFO/ForkPoolWorker-3] reason was: Waiting for next available executor on ‘huge&&x86_64&&bionic’
[2019-08-13 19:46:04,395: WARNING/ForkPoolWorker-3] invalid syntax (<unknown>, line 1)
[2019-08-13 19:46:04,395: WARNING/ForkPoolWorker-3] unable to match: ‘huge&&x86_64&&bionic’
[2019-08-13 19:46:04,396: WARNING/ForkPoolWorker-3] will look at node labels and attempt a match
[2019-08-13 19:46:04,429: INFO/ForkPoolWorker-5] found Jenkins nodes added by this service: 8
[2019-08-13 19:46:04,438: WARNING/ForkPoolWorker-3] "huge&&x86_64&&bionic" was not found in Jenkins
[2019-08-13 19:46:04,440: WARNING/ForkPoolWorker-3] unable to match a suitable node
[2019-08-13 19:46:04,440: WARNING/ForkPoolWorker-3] will infer from job labels
[2019-08-13 19:46:04,444: INFO/ForkPoolWorker-5] 167.114.233.191+xenial_trusty_pbuilder_huge__b2855ca6-686b-4fca-bf55-5090b616a90c is not idle, reset node.idle_since
[2019-08-13 19:46:04,449: ERROR/ForkPoolWorker-3] Task async.check_queue[710c5b5a-eae4-4ec9-a8b7-1cb6d3d9382b] raised unexpected: JenkinsException(u'job[ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge] does not exist',)
Traceback (most recent call last):
  File "/opt/mita/local/lib/python2.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/opt/mita/local/lib/python2.7/site-packages/celery/app/trace.py", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/mita/src/mita/mita/async.py", line 190, in check_queue
    conn.get_job_info(job_name)
  File "/opt/mita/local/lib/python2.7/site-packages/jenkins/__init__.py", line 473, in get_job_info
    raise JenkinsException('job[%s] does not exist' % name)
JenkinsException: job[ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge] does not exist
```

**Comment by djgalloway** [comment]
This is definitely deployed

https://github.com/ceph/mita/blob/master/deploy/playbooks/roles/common/templates/prod_nodes.py.j2#L69-L80

So I'm not sure why it says it can't match a node.

**Comment by alfredodeza** [comment]
I am not sure what the issue is here, could you elaborate? I understand the exception, and we are logging these, and it looks like normal behavior when checking the queue by poking in jobs

**Comment by djgalloway** [comment]
I'm wondering if there needs to be an extra replacement here without the `\s`

https://github.com/ceph/mita/blob/master/mita/label_eval.py#L61

**Comment by djgalloway** [comment]
I see two symptoms.  I'm not sure if they're different issues.

**First Symptom/Issue**
```
[2019-08-13 20:03:05,058: WARNING/ForkPoolWorker-3] invalid syntax (<unknown>, line 1)
[2019-08-13 20:03:05,058: WARNING/ForkPoolWorker-3] unable to match: ‘huge&&x86_64&&bionic’
```

That shouldn't happen because we have a node defined in prod_nodes.py with those labels.  At other points in the log I see mita successfully reading labels in `centos || trusty` format, for example.  Is there a spacing/syntax issue?

**Second Symptom/Issue**
```
[2019-08-13 20:03:35,346: WARNING/ForkPoolWorker-4] unable to match a suitable node
[2019-08-13 20:03:35,347: WARNING/ForkPoolWorker-4] will infer from job labels
[2019-08-13 20:03:35,357: ERROR/ForkPoolWorker-4] Task async.check_queue[2e69c97b-fce8-4c25-98f1-39678e322776] raised unexpected: JenkinsException(u'job[ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge] does not exist',)
Traceback (most recent call last):
  File "/opt/mita/local/lib/python2.7/site-packages/celery/app/trace.py", line 385, in trace_task
    R = retval = fun(*args, **kwargs)
  File "/opt/mita/local/lib/python2.7/site-packages/celery/app/trace.py", line 648, in __protected_call__
    return self.run(*args, **kwargs)
  File "/opt/mita/src/mita/mita/async.py", line 190, in check_queue
    conn.get_job_info(job_name)
  File "/opt/mita/local/lib/python2.7/site-packages/jenkins/__init__.py", line 473, in get_job_info
    raise JenkinsException('job[%s] does not exist' % name)
JenkinsException: job[ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge] does not exist
```

The job name is `ceph-dev-new-build/ARCH=x86_64,AVAILABLE_ARCH=x86_64,AVAILABLE_DIST=bionic,DIST=bionic,MACHINE_SIZE=huge`

I suspect this (https://github.com/ceph/mita/blob/master/mita/util.py#L21) is stripping the `ceph-dev-new-build/` portion needed to look up the labels.

**Comment by alfredodeza** [comment]
> I'm wondering if there needs to be an extra replacement here without the `\s`
> 
> https://github.com/ceph/mita/blob/master/mita/label_eval.py#L61

That is actually "zero or more" so it *should* be equivalent to "without the `\s`"

**Comment by djgalloway** [comment]
I mean I think there's an issue with spacing.  Like it will recognize `foo && bar` but not `foo&&bar`.

Not that there are multiple `&&`'s.

**Comment by djgalloway** [comment]
Oh, I see;  You were correcting my assumption.  I still wonder what the first symptom/issue is about.

**Comment by alfredodeza** [comment]
I've found the issue with the first problem, I will follow up in a new issue because it is separate from the fix already provided in pr #130 . The parsing needs to log a bit better, and we need to cleanup the string from the unicode chars:

```
In [15]: print u'\u2018huge&&x86_64&&bionic\u2019'
‘huge&&x86_64&&bionic’

In [16]: label_eval.validate_and_parse(u'\u2018huge&&x86_64&&bionic\u2019')
WARNING:label_eval:invalid syntax (<unknown>, line 1)
Out[16]: set()
```

**Comment by alfredodeza** [comment]
It is hard to catch, but the single quotes are actually utf-8 chars: \u2018 and \u2019
